### PR TITLE
fix: Only sublist items when they have greater indentation

### PR DESCRIPTION
**Pull request by pksunkara**
Before this PR the following text will have `1` as root list item and `2` as sublist item.

```
 + 1
+ 2
```

Now, both `1` and `2` are root list items

### CODE REVIEW

**Comment by pksunkara** [comment]
Tests will be added in apiaryio/markdown-parser

File: src/markdown.c [hunk 1]
```diff
@@ -1744,7 +1742,7 @@ parse_listitem(struct buf *ob, struct sd_markdown *rndr, uint8_t *data, size_t s
 			if (in_empty)
 				has_inside_empty = 1;
 
-			if (pre == orgpre) /* the following item must have */
+			if (i <= orgpre) /* the following item must have */
```
Comment (klokane) [inline src/markdown.c]: why this change?

Comment (pksunkara) [inline src/markdown.c]: That is what I explained in the main comment of this PR. List items with lesses indentation should not be a sublist.

Comment (pksunkara) [inline src/markdown.c]: Also from slack:

> Basically, the `==` means, the next line has the same indentation, so, it’s a next list item. But in our test case, the next line is less indented which means it doesn’t satisfy this condition and thus becomes a sublist which is wrong

Comment (klokane) [inline src/markdown.c]: what about something like this:
```
 + a
+ b
     + c
    + d
```

Comment (pksunkara) [inline src/markdown.c]: It would be parsed as:

```
+ root (type 0, data 0) - `` :0:26
  + list item (type 6, data 0) - `` :0:5
    + paragraph (type 7, data 0) - `a
` :3:2
  + list item (type 6, data 0) - `` :5:21
    + paragraph (type 7, data 0) - `b
` :7:2
    + list item (type 6, data 0) - `` :13:5
      + paragraph (type 7, data 0) - `c
` :16:2
    + list item (type 6, data 0) - `` :22:4
      + paragraph (type 7, data 0) - `d
` :24:2
```

Which is correct.

Comment (pksunkara) [inline src/markdown.c]: Why are you worried about that? You have a `break` in the next line. So, any other text after this doesn't need to be talked about.

Comment (klokane) [inline src/markdown.c]: Just to be sure. I try to go deep in logic. It seems correct.
